lib/archethic.ex

```diff
@@ -69,26 +56,41 @@ defmodule Archethic do
         welcome_node_key \\ Crypto.first_node_public_key()
       ) do
     if P2P.authorized_and_available_node?() do
-      do_send_transaction(tx, welcome_node_key)
+      case P2P.verify_synchronization() do
+        {:ok, :node_in_sync} ->
+          do_send_transaction(tx, welcome_node_key)
+
+        {:error, :node_out_of_sync} ->
+          forward_transaction(tx)
+
+        {:error, last_address_to_sync} ->
+          Resync.node_out_of_sync(last_address_to_sync)
+          forward_transaction(tx)
+      end
     else
-      P2P.authorized_and_available_nodes()
-      |> Enum.filter(&Node.locally_available?/1)
-      |> P2P.nearest_nodes()
-      |> forward_transaction(tx)
+      # node not authorized
+      forward_transaction(tx)
     end
   end
 
-  defp forward_transaction([node | rest], tx) do
+  defp forward_transaction(
+         tx,
+         nodes \\ P2P.authorized_and_available_nodes()
+         |> Enum.filter(&Node.locally_available?/1)
+         |> P2P.nearest_nodes()
+       )
+
+  defp forward_transaction(tx, [node | rest]) do
     case P2P.send_message(node, %NewTransaction{transaction: tx}) do
```

Comment:
When a node forward a transaction, the destination node will use the sender of the `NewTransaction` message as the welcome node, but there is some case where we can chain multiple forward. If so, the original welcome node will be loosed.
We should add the original welcome node in the `NewTransaction` message so in all case, the right welcome node will be provided to the validation nodes